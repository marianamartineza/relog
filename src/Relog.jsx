import PropTypes from 'prop-types';
import { useState } from 'react';

export const Relog = ({ hora, min, tiempo }) => {
  const [minute, setMinute] = useState(min);
  const [hour, setHour] = useState(hora);
  const [time, setTime] = useState(tiempo);

  const AddHour = () => {
    if (hour < 11) {
      setHour(hour + 1);
    } else if (hour == 11) {
      setHour(hour + 1);
      Evaluar(time);
    } else {
      // cuando sean las 12 cambiar a 1
      setHour(1);
    }
  };

  const AddMinute = () => {
    if (minute < 59) {
      setMinute(minute + 1);
    } else {
      setMinute(0);
      AddHour();
    }
  };

  const Evaluar = time => {
    if (time == 'AM') {
      setTime('PM');
    } else if (time == 'PM') {
      setTime('AM');
    }
  };

  const reset = () => {
    setHour(12);
    setMinute(0);
    setTime('AM');
  };

  return (
    <>
      <p data-testid='test-hour'>
        {hour}:{minute} {time}
      </p>
      <button aria-label='btn-hora' onClick={AddHour}>
        Hora
      </button>
      <button aria-label='btn-minuto' onClick={AddMinute}>
        Minutos
      </button>
      <button aria-label='btn-reset' onClick={reset}>
        Reiniciar
      </button>
    </>
  );
};

Relog.propTypes = {
  hora: PropTypes.number.isRequired,
  min: PropTypes.number.isRequired,
  tiempo: PropTypes.string.isRequired,
};

Relog.defaultProps = {
  hora: 12,
  min: 0,
  tiempo: 'AM',
};
