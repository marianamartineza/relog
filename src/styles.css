html,
body {
  background-color: black;
  font-size: 2rem;
  color: white;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

button {
  padding: 15px;
  font-size: 1.3rem;
  background-color: aqua;
  margin-right: 5px;
}
