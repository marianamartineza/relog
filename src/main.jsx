import React from 'react';
import ReactDOM from 'react-dom/client';
import { Relog } from './Relog';
import './styles.css';
// import {test} from './shared/utils';

// const App = () => console.log('prueba react');

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <Relog hora={0} min={0} tiempo="AM"/> */}
    <Relog hora={11} min={59} tiempo='PM' />
  </React.StrictMode>
);
