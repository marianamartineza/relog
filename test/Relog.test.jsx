
import { render, screen } from "@testing-library/react";
import { Relog } from "../src/Relog";



describe('pruebas en <Relog/>', () => { 
    
    test('debe hacer match con el snapshot', () => { 
        const {container}=render(<Relog/>);
        expect(container).toMatchSnapshot();
     });

     


 });