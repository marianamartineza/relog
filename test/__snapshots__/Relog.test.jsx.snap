// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <Relog/> debe hacer match con el snapshot 1`] = `
<div>
  <p
    data-testid="test-hour"
  >
    12
    :
    0
     
    AM
  </p>
  <button
    aria-label="btn-hora"
  >
    Hora
  </button>
  <button
    aria-label="btn-minuto"
  >
    Minutos
  </button>
  <button
    aria-label="btn-reset"
  >
    Reiniciar
  </button>
</div>
`;
